fe/fe-core/src/main/java/org/apache/doris/planner/external/TVFScanNode.java

```diff
@@ -70,6 +74,22 @@ protected void doInitialize() throws UserException {
         initSchemaParams();
     }
 
+    @Override
+    protected void initBackendPolicy() throws UserException {
+        List<String> preLocations = new ArrayList<>();
+        if (tableValuedFunction instanceof LocalTableValuedFunction) {
+            // For local tvf, the backend was specified by backendId
+            Long backendId = ((LocalTableValuedFunction) tableValuedFunction).getBackendId();
+            Backend backend = Env.getCurrentSystemInfo().getBackend(backendId);
+            if (backend == null) {
+                throw new UserException("No available backends");
+            }
+            preLocations.add(backend.getHost());
```

Comment:
better to rename `preLocations` to `preferLocations`. This is not your problem, but you can change it in this PR.